spec/classes/letsencrypt_spec.rb

```diff
@@ -25,11 +27,55 @@
                                                                       manage_dependencies: true,
                                                                       repo: 'https://github.com/certbot/certbot.git',
                                                                       version: 'v0.9.3').that_notifies('Exec[initialize letsencrypt]')
-
-            is_expected.to contain_ini_setting('/etc/letsencrypt/cli.ini email foo@example.com')
-            is_expected.to contain_ini_setting('/etc/letsencrypt/cli.ini server https://acme-v01.api.letsencrypt.org/directory')
             is_expected.to contain_exec('initialize letsencrypt')
             is_expected.to contain_class('letsencrypt::config').that_comes_before('Exec[initialize letsencrypt]')
+
+            case facts[:operatingsystem]
+            when 'FreeBSD'
+              is_expected.to contain_ini_setting('/usr/local/etc/letsencrypt/cli.ini email foo@example.com')
+              is_expected.to contain_ini_setting('/usr/local/etc/letsencrypt/cli.ini server https://acme-v01.api.letsencrypt.org/directory')
+            else
+              is_expected.to contain_ini_setting('/etc/letsencrypt/cli.ini email foo@example.com')
+              is_expected.to contain_ini_setting('/etc/letsencrypt/cli.ini server https://acme-v01.api.letsencrypt.org/directory')
+            end
+
+            if facts[:osfamily] == 'RedHat' && facts[:operatingsystemmajrelease] == '7'
+              is_expected.to contain_class('epel').that_comes_before('Package[letsencrypt]')
+              is_expected.to contain_class('letsencrypt::install').with(install_method: 'package')
+              is_expected.to contain_class('letsencrypt').with(package_command: 'certbot')
+              is_expected.to contain_package('letsencrypt').with(name: 'certbot')
+              is_expected.to contain_file('/etc/letsencrypt').with(ensure: 'directory')
+            elsif facts[:operatingsystem] == 'Debian' && facts[:operatingsystemmajrelease] == '8'
+              is_expected.to contain_class('letsencrypt::install').with(install_method: 'package')
+              is_expected.to contain_file('/etc/letsencrypt').with(ensure: 'directory')
+            elsif facts[:operatingsystem] == 'Ubuntu' && facts[:operatingsystemmajrelease] == '14.04'
+              is_expected.to contain_class('letsencrypt::install').with(install_method: 'vcs')
+              is_expected.to contain_file('/etc/letsencrypt').with(ensure: 'directory')
+            elsif facts[:operatingsystem] == 'Ubuntu' && facts[:operatingsystemmajrelease] == '16.04'
```

Comment:
You can leave out the second part of the condition and drop the 18.04 case below as well.